src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1030,49 +1069,61 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
         protected abstract INamedTypeSymbol CommonCreateTupleTypeSymbol(
             ImmutableArray<ITypeSymbol> elementTypes,
             ImmutableArray<string> elementNames,
-            ImmutableArray<Location> elementLocations);
+            ImmutableArray<Location> elementLocations,
+            ImmutableArray<NullableAnnotation> elementNullableAnnotations);
 
 #pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and
+        /// (optional) element names, locations, and nullable annotations.
+        /// The underlying type needs to be tuple-compatible.
         /// </summary>
-        /// <remarks>
-        /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
-        /// Also, the underlying type needs to be tuple-compatible.
-        /// </remarks>
         public INamedTypeSymbol CreateTupleTypeSymbol(
             INamedTypeSymbol underlyingType,
-            ImmutableArray<string> elementNames = default(ImmutableArray<string>),
-            ImmutableArray<Location> elementLocations = default(ImmutableArray<Location>))
+            ImmutableArray<string> elementNames = default,
+            ImmutableArray<Location> elementLocations = default,
+            ImmutableArray<NullableAnnotation> elementNullableAnnotations = default)
         {
             if ((object)underlyingType == null)
             {
                 throw new ArgumentNullException(nameof(underlyingType));
             }
 
-            return CommonCreateTupleTypeSymbol(
-                underlyingType, elementNames, elementLocations);
+            return CommonCreateTupleTypeSymbol(underlyingType, elementNames, elementLocations, elementNullableAnnotations);
         }
 #pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
 
+        /// <summary>
+        /// Returns a new INamedTypeSymbol with the given underlying type and element names and locations.
+        /// The underlying type needs to be tuple-compatible.
+        /// </summary>
+        /// <remarks>This overload is for backwards compatibility. Do not remove.</remarks>
+        public INamedTypeSymbol CreateTupleTypeSymbol(
```

Comment:
>CreateTupleTypeSymbol [](start = 32, length = 21)

Just to confirm: we have no way of testing this overload, correct?

Reply:
A call with exactly three arguments will use this overload.

---
In reply to: [302725701](https://github.com/dotnet/roslyn/pull/37048#discussion_r302725701) [](ancestors = 302725701)